sw/device/lib/base/memory.c

```diff
@@ -76,24 +212,85 @@ int OT_PREFIX_IF_NOT_RV32(memrcmp)(const void *lhs, const void *rhs,
 
 OT_WEAK
 void *OT_PREFIX_IF_NOT_RV32(memchr)(const void *ptr, int value, size_t len) {
-  uint8_t *ptr8 = (uint8_t *)ptr;
-  uint8_t value8 = (uint8_t)value;
-  for (size_t i = 0; i < len; ++i) {
-    if (ptr8[i] == value8) {
-      return ptr8 + i;
+  const unsigned char *ptr8 = (const unsigned char *)ptr;
+  const uint8_t value_byte = (uint8_t)value;
+
+  size_t leading_bytes_end, aligned_words_end;
+  compute_alignment((uintptr_t)ptr, (uintptr_t)NULL, len, &leading_bytes_end,
+                    &aligned_words_end);
+  size_t index = 0;
+  for (; index < leading_bytes_end; ++index) {
+    if (ptr8[index] == value_byte) {
+      return (void *)&ptr8[index];
```

Comment:
nit: (pointer arithmetic should be fine on `unsigned char *`, or no?) also `void *` does not need a cast.
```suggestion
      return ptr8 + i;
```

Reply:
The dirty secret here is that `(void *)` is discarding the `const` qualifier on the `ptr` parameter. It's kind of a footgun, but it's also the type signature of the function...

Reply:
Ah, I see. Thanks for the explanation!